src/builder/command.rs

```diff
@@ -4140,15 +4155,40 @@ impl Command {
 
         if !self.is_set(AppSettings::DisableHelpSubcommand) {
             debug!("Command::_check_help_and_version: Building help subcommand");
-            let mut help_subcmd = Command::new("help")
-                .about("Print this message or the help of the given subcommand(s)")
-                .arg(
+            let help_about = "Print this message or the help of the given subcommand(s)";
+
+            let mut help_subcmd = if self.is_set(AppSettings::ExpandHelpSubcommandTrees) {
+                // Slow code path to recursively clone all other subcommand subtrees under help
+                let help_subcmd = Command::new("help")
+                    .about(help_about)
+                    .global_setting(AppSettings::DisableHelpSubcommand)
+                    .subcommands(self.get_subcommands().cloned().map(|mut sc| {
+                        // Remove args so help completion will not suggest them, only subcommands
+                        sc._clear_args_recursive();
+
```

Comment:
I'm concerned there might be some settings on these commands that produces unexpected results.

Should we instead create new commands, only copying over what we want?